src/ucp/tag/rndv.c

```diff
@@ -19,6 +19,29 @@ static int ucp_rndv_is_get_zcopy(ucp_request_t *sreq, ucp_rndv_mode_t rndv_mode)
              UCP_MEM_IS_HOST(sreq->send.mem_type)));
 }
 
+static int ucp_rndv_is_pipeline_needed(ucp_request_t *sreq) {
+    ucp_rndv_mode_t rndv_mode;
+    uct_md_attr_t *md_attr;
+    unsigned md_index;
+
+    rndv_mode = sreq->send.ep->worker->context->config.ext.rndv_mode;
+    md_index  = ucp_ep_md_index(sreq->send.ep, sreq->send.lane);
+    md_attr   = &sreq->send.ep->worker->context->tl_mds[md_index].attr;
+
+    if ((rndv_mode == UCP_RNDV_MODE_PUT_ZCOPY) ||
+        ((rndv_mode == UCP_RNDV_MODE_AUTO) &&
+         UCP_MEM_IS_HOST(sreq->send.mem_type))) {
```

Comment:
minor: i would check  UCP_MEM_IS_HOST(sreq->send.mem_type) first (even right at the beginning of the function) because it's the fast-path case